drivers/storage/portworx/component/prometheus.go

```diff
@@ -173,6 +183,27 @@ func (c *prometheus) MarkDeleted() {
 	c.isOperatorCreated = false
 }
 
+// createPrometheusCRDs registers all CRDs needed by prometheus operator
+// with prometheus operator upgraded to 0.50.0, it no longer registers CRDs automatically
+// check for details: https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/crds
+func (c *prometheus) createPrometheusCRDs() error {
+	filename := path.Join(pxutil.SpecsBaseDir(), "prometheus-crd-*.yaml")
+	files, err := filepath.Glob(filename)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		if err := k8sutil.ParseObjectFromFile(file, c.scheme, crd); err != nil {
+			return err
+		}
+		if err := k8sutil.CreateCRD(crd); err != nil {
```

Comment:
nit: Ideally we should optimize this further by first checking if the CRD is already present and then reading+parsing+creating the CRD. Will save some compute and i/o cycles.
You could address this now or just add a TODO and pick it up later.

Reply:
but we get the crd list from files, only after that we fetch the crd names, then check if it exists. so need to parse first anyways.

Reply:
Ah, good point! May be we can do some optimization in the future with the file name being the crd name. Anyway, it is okay for now.